extensions/wikia/AdEngine/js/video/ooyalaAdSetProvider.js

```diff
@@ -0,0 +1,75 @@
+/*global define*/
+define('ext.wikia.adEngine.video.ooyalaAdSetProvider', [
+	require.optional('ext.wikia.adEngine.adContext'),
+	require.optional('ext.wikia.adEngine.video.vastUrlBuilder')
+], function (adContext, vastUrlBuilder) {
+	'use strict';
+
+	var FEATURED_VIDEO_RATIO = 640 / 480;
+
+	function generateSet(vpos, type, position, rv, correlator) {
+		return {
+			position_type: type,
+			position: position,
+			tag_url: vastUrlBuilder.build(FEATURED_VIDEO_RATIO, {
+				pos: 'FEATURED',
+				src: 'premium',
+				rv: rv
+			}, {
+				vpos: vpos,
+				correlator: correlator
+			})
+		};
+	}
+
+	function calculateRV(videoDepth, capping) {
+		return videoDepth === 1 ? 1 : Math.floor((videoDepth - 1) / capping) + 1;
+	}
+
+	function shouldPlayNextVideoAd(videoDepth, capping) {
+		return (videoDepth - 1) % capping === 0;
+	}
+
+	function isAbleToDisplayAds() {
+		return adContext && adContext.getContext() && adContext.getContext().opts.showAds && vastUrlBuilder;
+	}
+
+	function adsCanBeplayedOnNextVideoViews() {
+		return adContext && adContext.getContext() && adContext.getContext().opts.replayAdsForFV;
+	}
+
+	function get(videoDepth, correlator) {
+		if (!isAbleToDisplayAds()) {
+			return [];
+		}
+
+		videoDepth = videoDepth !== undefined ? videoDepth : 1;
```

Comment:
can we reverse this `if` to : `videoDepth = videoDepth || 1` ? discussed offline that videoDepth should never be 0